src/ray/core_worker/core_worker.cc

```diff
@@ -779,24 +781,27 @@ Status CoreWorker::CreateActor(const RayFunction &function,
       actor_creation_options.is_direct_call, actor_creation_options.max_concurrency,
       actor_creation_options.is_detached, actor_creation_options.is_asyncio);
 
-  std::unique_ptr<ActorHandle> actor_handle(
-      new ActorHandle(actor_id, job_id, /*actor_cursor=*/return_ids[0],
-                      function.GetLanguage(), actor_creation_options.is_direct_call,
-                      function.GetFunctionDescriptor(), extension_data));
-  RAY_CHECK(AddActorHandle(std::move(actor_handle)))
-      << "Actor " << actor_id << " already exists";
-
   *return_actor_id = actor_id;
   TaskSpecification task_spec = builder.Build();
+  Status status;
   if (actor_creation_options.is_direct_call) {
     task_manager_->AddPendingTask(
         GetCallerId(), rpc_address_, task_spec,
         std::max(RayConfig::instance().actor_creation_min_retries(),
                  actor_creation_options.max_reconstructions));
-    return direct_task_submitter_->SubmitTask(task_spec);
+    status = direct_task_submitter_->SubmitTask(task_spec);
   } else {
-    return local_raylet_client_->SubmitTask(task_spec);
+    status = local_raylet_client_->SubmitTask(task_spec);
   }
+
+  *actor_object_id = return_ids[0];
```

Comment:
Is it true that this value will always be the same as `ObjectID::ForTaskReturn(TaskID::ForActorCreationTask(actor_id), 0)`? I didn't see any random bits in this.

If so, maybe we can generate the id on the fly instead of storing it.

Reply:
Oh hmm yeah I didn't see that function. That should work for C++.

We'll still have to store it in Python, though, since we're relying on the `ObjectID`'s python ref to track when the local ref goes out of scope and when the handle gets passed into a task.